pkg/healthz/clock_health.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/utils/clock"
+)
+
+type clockHealthAdaptor struct {
+	clock              clock.Clock
+	startTimeReal      time.Time
+	startTimeMonotonic time.Time
+}
```

Comment:
suggestion: I think this check is _incredibly_ subtle and most people would be bewildered trying to work out why this exists. This health check would really benefit from a lot of doc comments explaining why it's needed.

Specifically, I think the implementation in this file talks a lot about "real time" vs "monotonic time" and it's not really clear that the ultimate problem this PR addresses is that on Linux systems, monotonic time can stop in certain situations. If I see "monotonic time" without futher context I'm thinking there's gonna be an issue with timezones or leap seconds or something!

Reply:
Indeed, I added a comment above this clockHealthAdaptor struct that explains why this health check exists.